modules/cluster-management/src/main/java/org/apache/ignite/internal/cluster/management/ClusterManagementGroupManager.java

```diff
@@ -350,6 +361,49 @@ public CompletableFuture<Void> startAsync(ComponentContext componentContext) {
         return nullCompletedFuture();
     }
 
+    private CompletableFuture<Void> doClusterReset(ResetClusterMessage resetClusterMessage) {
+        LOG.info("Found a ResetClusterMessage in storage, going to do cluster reset [message={}]", resetClusterMessage);
+
+        return destroyCmgWithEvents()
+                .thenCompose(unused -> {
+                    if (resetClusterMessage.cmgNodes().contains(clusterService.nodeName())) {
+                        return doReinit(resetClusterMessage);
+                    } else {
+                        // Let's just wait for new CMG nodes to establish a majority and send us an invitation to join.
+                        cmgMessageHandler.onRecoveryComplete();
+                        return nullCompletedFuture();
+                    }
+                })
+                .thenRun(clusterResetStorage::removeResetClusterMessage);
+    }
+
+    private CompletableFuture<Object> doReinit(ResetClusterMessage resetClusterMessage) {
+        CompletableFuture<CmgRaftService> serviceFuture;
+
+        synchronized (raftServiceLock) {
+            // Disaster recovery means that the Repair Conductor has ensured the cluster was initialized,
+            // so we can just pass null as initialClusterConfig.
+            serviceFuture = startCmgRaftServiceWithEvents(
```

Comment:
I wonder if starting starting the service with an empty config is the correct thing: we will produce a fake event in this case, can it break something?

Reply:
I'm not sure what a 'fake event' is, could you please elaborate?

A CMG repair is only allowed if the initial configuration is already in the Metastorage, so, if we are executing this code, this means that the initial configuration (if it existed) is already applied and it will never be accessed in the CMG. That's why we can pass `null` here.

Reply:
ok, I just want to be sure that components, that consume this event, will handle this situation correctly